cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/DeduplicateAggregationDefinition.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Specifies how a deduplicate operation should be executed.
+ * A deduplication is an operation that takes in a set of rows and for each group of rows for which the values for
+ * certain group by fields are the same, outputs a single row.
+ * This single row can be determined by performing an operation on the set of rows which returns a single row.
```

Comment:
Could you expand on this? Basically filterExpressions define a composite key and a row with the specified (minimum or maximum according to the function for the column) key is selected

Reply:
A single filterExpressions pair defines a single key and a function (MIN or MAX). If this function returns more than one row, then the next filterExpressions pair is used and so on.

Reply:
I mean put better explanation in the javadoc, so that a person who does not know how dedup works would understand it

Reply:
I know, I did that too :)